vars/piperExecuteBin.groovy

```diff
@@ -0,0 +1,83 @@
+import com.sap.piper.JenkinsUtils
+import com.sap.piper.PiperGoUtils
+import com.sap.piper.Utils
+
+import static com.sap.piper.Prerequisites.checkScript
+
+void call(Map parameters = [:], stepName, metadataFile, List credentialInfo, failOnMissingReports = false, failOnMissingLinks = false) {
+
+    handlePipelineStepErrors(stepName: stepName, stepParameters: parameters) {
+
+        def script = checkScript(this, parameters) ?: this
+        parameters.script = null
+
+        def utils = parameters.juStabUtils ?: new Utils()
+        parameters.juStabUtils = null
+
+        def jenkinsUtils = parameters.jenkinsUtilsStub ?: new JenkinsUtils()
+        parameters.jenkinsUtilsStub = null
+
+        new PiperGoUtils(this, utils).unstashPiperBin()
+        utils.unstash('pipelineConfigAndTests')
+        script.commonPipelineEnvironment.writeToDisk(script)
+
+        writeFile(file: metadataFile, text: libraryResource(metadataFile))
```

Comment:
I would like to suggest to write the file into the `.pipeline` folder somewhere. We should not have serveral folders for some kind of auxiliary files. The best approach IMO would be to have some folder inside `.pipeline` receiving additional files - just as an example `./pipeline/tmp`.

Here in this case we would come into trouble when the project comes with some `metadata` folder. It might happen that - by the project - the metadata folder is zipped and put somewhere into the deliverable.

Reply:
good point!
This goes hand in hand with the earlier discussions we had around `.gitignore` for temporary files created during running the binary.

Will look into it ...

Reply:
Just updated it accordingly ...